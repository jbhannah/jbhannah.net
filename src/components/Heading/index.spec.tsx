import { render, shallow } from "enzyme"
import * as React from "react"
import { Heading, HeadingFactory } from "."

const TestHeading = (props) => shallow(<Heading {...props} />)

describe("Heading", () => {
  describe("with a valid heading level", () => {
    const tree = TestHeading({ level: "h1" })

    it("renders correctly", () => {
      expect(tree).toMatchSnapshot()
    })
  })
})

describe("HeadingFactory", () => {
  const H1 = HeadingFactory("h1")
  const tree = render(<H1></H1>)

  it("returns a Heading", () => {
    expect(tree.is("h1")).toBeTruthy()
  })
})
