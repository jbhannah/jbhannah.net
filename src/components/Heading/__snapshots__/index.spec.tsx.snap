// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading with a valid heading level renders correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Heading"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="h1"
  css={
    Object {
      "a": Object {
        "background": "none",
        "textShadow": "none",
      },
      "code": Object {
        "fontSize": "inherit",
      },
    }
  }
/>
`;
